misc/multiaddr/src/lib.rs

```diff
@@ -29,17 +29,17 @@ pub use self::protocol::Protocol;
 
 /// Representation of a Multiaddr.
 #[derive(PartialEq, Eq, Clone, Hash)]
-pub struct Multiaddr { bytes: Bytes }
+pub struct Multiaddr { bytes: BytesMut }
```

Comment:
Cloning `Bytes` is cheap, cloning `BytesMut` not so much. If we move away from `Bytes` we might as well use a `Vec<u8>` as we did in the past.

Reply:
I don't know much about memory cost of either. Would you like me to change to `Vec<u8>` on this PR?

Reply:
No, I think we should leave the implementation as is for the time being and not upgrade to bytes-0.5. Eventually we will probably want a custom copy-on-write implementation with cheap clones for shared `Multiaddr` values.